System/FilePath/Internal.hs

```diff
@@ -313,6 +313,15 @@ hasExtension :: FilePath -> Bool
 hasExtension = any isExtSeparator . takeFileName
 
 
+-- | Is the given string the final extension of the filename?
+--   The extension should not include the separator.
+--
+-- > "png" `isExtensionOf` "/directory/file.png"     == True
```

Comment:
Oh yeah, I considered it. It's certainly prettier than the == approach, but I noticed that the rest of the module used that so I went along with it.